service/src/main/kotlin/fi/espoo/evaka/assistanceneed/AssistanceNeedController.kt

```diff
@@ -54,7 +59,23 @@ class AssistanceNeedController(
     ): ResponseEntity<List<AssistanceNeed>> {
         Audit.ChildAssistanceNeedRead.log(targetId = childId)
         accessControl.requirePermissionFor(user, Action.Child.READ_ASSISTANCE_NEED, childId)
-        return assistanceNeedService.getAssistanceNeedsByChildId(db, childId).let(::ok)
+        return assistanceNeedService.getAssistanceNeedsByChildId(db, childId).filter { assistanceNeed ->
+            // If child is or has been in preschool, do not show pre preschool daycare assistance needs for non admin
+            if (!user.isAdmin) {
```

Comment:
Perhaps something like `if (accessControl.hasPermissionFor(user, Action.Child.READ_ASSISTANCE_NEED_FULL, childId)` would be better than a hard-coded check for admin.

Reply:
Did a whole revamp by moving this stuff to access control side